dataset-spec/README.md

```diff
@@ -0,0 +1,86 @@
+| -------------- | ------- | --------------------- | ------------------------------------------------------------ |
+| description    | string  | Description           | Detailed description to explain the hosting details. [CommonMark 0.28](http://commonmark.org/) syntax MAY be used for rich text representation. |
+| scheme         | string  | Scheme (required)     | Values: S3, GCS, URL, OTHER                                  |
+| id             | string  | Identifier (required) | Host-specific identifier such as an URL or asset id.         |
+| region         | string  | Region                | Provider specific region where the data is stored.           |
+| requester_pays | boolean | Requester pays        | `true` if requester pays, `false` if host pays.              |
+
+**Note:** The idea of storage profiles is currently [discussed](https://github.com/radiantearth/stac-spec/issues/148). Therefore, scheme, id and region may be removed from the final spec.
+
+### Link Object
+
+TODO: Should be compatible with STAC or remove if catalog is revised.
+
+| Element | Type   | Name                           | Description                         |
+| ------- | ------ | ------------------------------ | ----------------------------------- |
+| href    | string | Hyperlink reference (required) |                                     |
+| rel     | string | Relation (required)            |                                     |
+| type    | string | MIME-type                      | MIME-type of the referenced entity. |
+| title   | string | Title                          | Human-readable title for the link.  |
+
+##  EO extension (eo) - Items and Datasets
+
+We follow the STAC EO extension where meaningful (see below), but propose additional fields:
+
+| Element      | Type     | Name               | Description           |
+| ------------ | -------- | ------------------ | --------------------- |
+| asset_schema | object   | Asset Schema       | TODO                  |
+| nodata       | [number] | Nodata values      | The no data value(s). |
+| pyramid      | object   | Pyramid parameters | TODO                  |
+| periodicity  | string   | Periodicity        | ISO8601               |
+
+Some fields such as `eo:sun_elevation ` or `eo:sun_azimuth` are only meaningful on the item level and SHOULD not be used in datasets.
+
+#### Bands
+
+We follow the STAC EO extension for bands, but propose additional fields:
+
+| Element | Type     | Name          | Description                                                  |
+| ------- | -------- | ------------- | ------------------------------------------------------------ |
+| nodata  | [number] | Nodata values | The no data value(s).                                        |
+| offset  | number   | Offset        | Offset to convert band values to the actual measurement scale. |
+| scale   | number   | Scale         | Scale to convert band values to the actual measurement scale. |
```

Comment:
Been thinking about this more and scale and offset definitely need to be specified per asset, not per band. Individual items can have different scale and offsets. While Landsat collection 1 uses a consistent scale/offset for all scenes, this was not always the case and a provider may certainly provide them per Item, it is not really a band characteristic, it is an asset characteristic.
I had thought of it because there are other reasons too - for instance I might take the RGB bands and scale them to  a Byte and provide that as a visible True Color image, or other bands to make a False Color image, but still want to reference the bands that make up that 3-band image.

Reply:
In the same vain, units really is per asset as well. I might provide a TOA datafile as well as a Surface reflectance file.

And in those cases I might want to use different nodata values as well, so they really should all be per-asset.  Units and nodata can be specified per asset in the Dataset, but scale/offset needs to be specified for each asset in each Item.

Reply:
@m-mohr @matthewhanson gain (scale) and offsets may have distinct values depending on the date, see the case for [LS4 and LS5](https://landsat.usgs.gov/april-25-2017-changes-l4-and-l5-radiometric-calibration-collection-1-processing). How should that cases be represented? I also prefer gain instead of scale, but this may be a Landsat thing.

Reply:
@matthewhanson : I don't quite follow. What is an asset in this case? My understanding that by assets STAC people usually mean files on disk. Do you mean a specific file for a specific item? But a gdal file can have two bands that have different gain/offset.

For reference, datasets in EE almost always have the same gain/offset for each band across all items. The two exceptions are Landsat and SeaWiFS OceanColor dataset that has different processing for different items in the same dataset. We did not dare to change Landsat, but for SeaWiFS we applied gain/offset before ingestion to make the data easier to use in ARD-like manner. 

So in general, yes, each item can have its own gain/offset. I've never seen nodata values differ across items, though. If they do, it might make more sense to split the dataset in two or more.

Reply:
@matthewhanson In the normal "STAC way" with Items and Assets that might be feasible, but for openEO (and GEE?), we still need those information - whenever meaningful - per dataset or band, as we don't have any items/assets where this information is available! 

@fredliporace 
- Honestly, it's just a name, but I am more used to scale. Don't really care whether its gain or scale. It has the same meaning, correct?
- Not sure whether we can cover all of that, currently it can't represented. Maybe just leave it out if it depends on another property? I don't have a proper solution for it yet - but open for proposals.